### DISCUSSION TITLE
Less confusing/more permanent behaviour for photos in shared albums

### DISCUSSION THREAD

**Discussion by ajsteele** [ente-io/ente#4285] [Enhancements]
If I understand correctly, the current behaviour when uploading a photo to a shared album that you do not own is to upload it to `Uncategorized` in your album, and add it to the shared album separately. ie, it seems as though the behaviour described here has already been implemented: https://github.com/ente-io/ente/discussions/811 Thus, photos you upload are ‘yours’, while other photos are ‘theirs’, and the only link between them is something in metadata.

I think there are a few problems with the current behaviour. :)

1. It's very confusing for non-tech-savvy users, and frankly took me a minute to get my head around! (as noted in the other discussion)
2. It will presumably lead to weird behaviour if albums are unshared in future…you might spend ages putting photos in shared albums, only for them to be unshared (possibly by accident!) and all your efforts categorising are lost, leaving just a mass of photos in `Uncategorized`, including every other photo you ever added to a shared album.
3. If someone unshares an album with you, you also lose all the photos they uploaded. I understand that in some cases this might be desired behaviour, but I would imagine in most uses cases, either
   - If sharing with someone untrusted, they could have downloaded all the pictures already anyway,
   - If sharing with a friend or family member, you might not want to revoke access.
5. I’ve not tried this, but presumably if someone stops using Ente in future and deletes their account, would all the photos they shared with you also be lost?
6. The recipient can’t do anything to change metadata on the shared album, eg album name, cover photo, sort order. Again, there could be cases where this is desirable, but often users presumably wouldn’t mind this!
7. The recipient’s operations on 

Some of this might be covered by Partner Sharing when that’s available, but I can imagine a fairly large set of shared album use-cases where these would be helpful (eg group events or holidays, friends you see often, etc), but I think in general the sharp distinction and wildly different functionality between a shared album you own and an album shared with you by someone is confusing and probably not what many end-users want.

I’m not sure what specifically to suggest because I know some of this is dictated by backend architecture, and I know there are lots of use-cases out there! But I’m trying hard to get non-techie friends using Ente and it’s tough when sharing pictures with them. Even finding the albums can be harder because they’re on a the Shared screen rather than Albums which is counter-intuitive somtimes. :)

Love the app and thanks for all your hard work!

**Comment by vishnukvmd** [comment]
Hey, thanks for the feedback! Thinking out loud,

1. One way forward would be to add items to a "shadow" album that shares the name of the shared album, instead of Uncategorized, and make it visible only when the uploader loses access to the shared album that they added items to.
2. "Shadow" albums would maintain the categorization, and pop up once you lose access to the original album.
3. This one is tricky, as a user I would want unshared items to disappear from the feed of folks who I shared them with. What we could do is provide an easy way for receivers to "save" all items from a shared album into their library.
4. Currently yes. "Save" could prevent this since on the backend it'll give us a user to attach the consumed storage to when the originals are deleted.
5. This is planned, we'll introduce a new permission level where a collaborator can act like an owner / admin.
6. I'm guessing the sentence got truncated? 👀 

Thanks again for the feedback, we'll be mindful of these points as we iterate on collaboration! :)

**Reply by ajsteele** [reply]
Thanks for the reply and sorry for the truncated item six! Teach me to write such a long suggestion. :)

I think I meant to write that the recipient’s actions on the shared items are really limited—basically you can only add them to another album, but can't eg move or delete. I find this a bit annoying, and I think less techie users would find the distinction confusing!

And thanks for replying to the other points—'shadow albums' might be a good way to do it, effectively a copy on both users' ends makes a lot of sense. I agree on 3 that in some cases it would be expected/desired behaviour for the sharing, but as I said above an adversarial user might have already downloaded them anyway, and in cases where you’re sharing with family or friends you might not want them to lose access—especially if the unshare was accidental! Maybe there should be a configurable default that can be overridden per-folder/user?

Thanks again for the responsiveness and looking forward to seeing how sharing evolves. :)
